third_party/nvfuser/csrc/scheduler/reduction_utils.cpp

```diff
@@ -140,7 +144,13 @@ TensorView* scheduleReductionTV(
       outer_unroll(outer_i++, rparams.unroll_factor_inner_reduction);
     }
 
-    reduction_tv->axis(outer_i)->parallelize(rparams.block_dim_inner_reduction);
+    if (rparams.combined_inner_outer && !rparams.multiple_reds_per_blk) {
```

Comment:
In the above, there's additional condition, `rparams.lparams.bdimx() > 1`. Why not used here?

Reply:
Also, the code would look easier to understand if there's only one place to use `rpams.block_dim_inner_reduction` (within this if block fo `rparams.persistent_kernel`).

Reply:
> In the above, there's additional condition, `rparams.lparams.bdimx() > 1`. Why not used here?

This condition shouldn't exist. I can't remember why it was there originally. Maybe I was doing some debugging.

Reply:
> Also, the code would look easier to understand if there's only one place to use `rpams.block_dim_inner_reduction` (within this if block fo `rparams.persistent_kernel`).

`rpams.block_dim_inner_reduction` is used to split the reduction dim by `NamedScalar::getParallelDim(ptype)` for combined reduction with single reduction per block. For regular reduction, it is only used to parallel the reduction dim. So it will appear twice in the code.

Reply:
I was wondering if this could also work:

```suggestion
    // delete the above use of block_dim_inner_reduction
    ...
    if (rparams.combined_inner_outer && !rparams.multiple_reds_per_blk) {
      inner_parallel(outer_i, rparams.block_dim_inner_reduction);
      reduction_tv->axis(outer_i)->parallelize(
          rparams.block_dim_inner_reduction_extra);
    } else {
       reduction_tv->axis(outer_i)->parallelize(
          rparams.block_dim_inner_reduction);
    }
```

This way, I think it's more apparent how we use `block_dim_inner_reduction` and `block_inner_dim_reduction_extra`.